distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/ConfigureTestCommons.java

```diff
@@ -13,71 +13,110 @@
+  private static final String METACARD_ATTRIBUTE_SECURITY_POLICY_PLUGIN_PID =
       "org.codice.ddf.catalog.security.policy.metacard.MetacardAttributeSecurityPolicyPlugin";
 
-  public static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
+  private static final String AUTH_Z_REALM_PID = "ddf.security.pdp.realm.AuthzRealm";
 
   private ConfigureTestCommons() {}
 
-  public static void configureMetacardValidityFilterPlugin(
-      List<String> securityAttributeMappings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID, null);
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("attributeMap", securityAttributeMappings);
-    config.update(properties);
+  public static Dictionary<String, Object> configureService(
+      String pid, Dictionary<String, Object> props, AdminConfig adminConfig) throws IOException {
+    Configuration config = adminConfig.getConfiguration(pid, null);
+    Dictionary<String, Object> oldProps = config.getProperties();
+    config.update(props);
+
+    // verify update
+    with()
+        .pollInterval(1, SECONDS)
+        .await()
+        .atMost(TimeUnit.MINUTES.toSeconds(2), SECONDS)
+        .ignoreExceptions()
+        .until(
+            () -> {
+              Dictionary<String, Object> savedProps = config.getProperties();
+              List<String> newKeys = Collections.list(props.keys());
+              for (String key : newKeys) {
+                Object savedValue = savedProps.get(key);
+                Object incomingValue = props.get(key);
+                if (incomingValue.getClass().isArray() && savedValue.getClass().isArray()) {
+                  if (!Arrays.equals((Object[]) savedValue, (Object[]) incomingValue)) {
+                    return false;
+                  }
+                } else if (!savedValue.equals(incomingValue)) {
+                  return false;
+                }
+              }
+              return true;
+            });
+
+    return oldProps;
   }
 
-  public static void configureFilterInvalidMetacards(
-      String filterErrors, String filterWarnings, AdminConfig configAdmin) throws IOException {
-    Configuration config =
-        configAdmin.getConfiguration(METACARD_VALIDITITY_FILTER_PLUGIN_SERVICE_PID, null);
-
+  // This method is intended to be used to configure the service for a test.
+  public static Dictionary<String, Object> configureMetacardValidityFilterPlugin(
+      List<String> securityAttributeMappings,
+      boolean filterErrors,
+      boolean filterWarnings,
+      AdminConfig configAdmin)
+      throws IOException {
     Dictionary<String, Object> properties = new DictionaryMap<>();
-    properties.put("filterErrors", filterErrors);
-    properties.put("filterWarnings", filterWarnings);
-    config.update(properties);
+    properties.put("attributeMap", securityAttributeMappings);
```

Comment:
✏️ Maybe use DictionaryMap.of(k1, v1, k2, v2, k3, v3)

Reply:
I'm not really a fan of how it reads passing multiple pairs in a single set of arguments